### ISSUE TITLE
Intermittent production error

### ISSUE THREAD

**Issue by StevenClontz** [IsToomersCornerBeingRolledRightNow/theAppThatLetsUsProcessToomersCornerBeingRolled#4]
I've been restarting the process by hand when this happens, but occasionally this error begins looping:

```
theapp.bin: user error (Pattern match failure in do expression at Canny.hs:62:5-23)
find: `tmp/': No such file or directory
ls: cannot access tmp/*: No such file or directory
```

The first time this appeared was after running without interruption for over a week. But the last two times I've checked, this has popped up. (I have a recurring reminder to check on the production app every three days.)

**Comment by friedbrice** [comment]
I'll look at it this afternoon.

**Comment by friedbrice** [comment]
are changes even allowed at this point?

**Comment by friedbrice** [comment]
I'm having trouble replicating the error on my laptop, but that makes sense since it's intermittent, so here's my best guess:

All three of these lines of console output

```
theapp.bin: user error (Pattern match failure in do expression at Canny.hs:62:5-23)
find: `tmp/': No such file or directory
ls: cannot access tmp/*: No such file or directory
```

are the result of the `tmp/` directory missing. (This is obvious for the last two, but for the first, the pattern match against the argument fails because the argument is empty. See line 62 of `Canny.hs`.)

In `controller.sh`, `find` (line 73) will delete any file in `tmp/` that is older than `$myCacheTime`. This applies to `tmp/` itself, but _only if `tmp/` is empty._ But `tmp/` should never actually empty, because `avconv` (line 36) is continually scraping frames off of the webcam and storing them in `tmp/`. We can "fix" the invocation of `find` so that it won't delete directories, but that won't address the bigger issue---something is killing `avconv`.

I'm presently looking into different solutions for how one can monitor (automatically) the activity of a forked process. There should be some kind of canonical UNIX tool for doing exactly that.

**Comment by StevenClontz** [comment]
@friedbrice I assume changes are allowed (judges can see what updates we've made since it's all tracked publicly on github and base their judgement accordingly). I'm more interested in developing the project further than the judging process anyway.

FWIW I'm using [GNU Screen](http://www.gnu.org/software/screen/) to run the script (actually a whole bash session) as a background process so I can easily SSH in, check on the running script, and then exit. I don't know if that's applicable to this situation.

**Comment by friedbrice** [comment]
oh, well then i'd better not run it as init scripts or daemons then. that
doesn't effect the current implementation, but it does effect the eventual
solution. thanks for mentioning.

**Comment by StevenClontz** [comment]
This hasn't popped up since, so I'm closing this.

**Comment by StevenClontz** [comment]
This is rearing its head again. Is it possible that a smaller virtual disk size (8GB vs 100GB) is causing the error? It appeared once I made that change (to put the app within the threshold for free service).

**Comment by friedbrice** [comment]
Is the disk filling up? My guess is it's not. Let me know.

**Comment by StevenClontz** [comment]
Actually it's close:

``` bash
~$ df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/xvda1      7.8G  7.3G  105M  99% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
udev            492M   12K  492M   1% /dev
tmpfs           100M  332K   99M   1% /run
none            5.0M     0  5.0M   0% /run/lock
none            497M     0  497M   0% /run/shm
none            100M     0  100M   0% /run/user
```

**Comment by friedbrice** [comment]
yeah, it's probably running out of space. how much space are we using for
pictures vs using for the OS?

**Comment by StevenClontz** [comment]
It had about 5GB of files in the images directory. I cleared it, and restarted the program. That might fix it - my theory is that when I swapped the 100GB back to the 8GB virtual disk, I forgot to clear out the cached images from the previous version which didn't clean up after itself (if I recall correctly).

**Comment by friedbrice** [comment]
Okay. We don't have to change any of the code, I can really just tweak a setting. That will control how long before I cashed image is deleted. I just had no idea the images were taken up so much space.

**Comment by friedbrice** [comment]
Well, I don't think it's just the old images, since it was working for a while before crapping out this time. I'll still make a change to how long it takes for a file to be delete it.

**Comment by StevenClontz** [comment]
I think it might be. The error appeared immediately after I switched to the smaller 8GB disk (which still had the old images from before). I'll let you know if caching is an issue when I check it again in a day ror so.

**Comment by StevenClontz** [comment]
``` bash
~$ du -sh
555M    .
ubuntu@ip-172-31-42-46:~$ screen -rD
./controller.sh: line 46: ./theapp.bin: No such file or directory
find: `tmp/': No such file or directory
ls: cannot access tmp/*: No such file or directory
```

Alright, so, it looks like hard disk space isn't an issue - we're well under capacity and this error is still occurring on occasion.

**Comment by friedbrice** [comment]
Steven, give me ssh access
